server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -4554,6 +4581,112 @@ public void syncSegmentsFromRemoteSegmentStore(boolean overrideLocal, boolean re
+
+                if (overrideLocal || localDirectoryContains(storeDirectory, file, checksum) == false) {
+                    if (localSegmentFiles.contains(file)) {
+                        storeDirectory.deleteFile(file);
+                    }
+                    if (remoteSegmentStoreDirectory != null) {
+                        String uploadedFileName = uploadedSegments.get(file).toString().split("::")[1];
+                        logger.debug(
+                            "local file is "
+                                + file
+                                + "uploaded file name is "
+                                + uploadedFileName
+                                + " and file is "
+                                + uploadedSegments.get(file).toString()
+                        );
+                        remoteSegmentStoreDirectory.copyFrom(
+                            tempRemoteSegmentDirectory,
+                            file,
+                            file,
+                            IOContext.DEFAULT,
+                            false,
+                            uploadedSegments.get(file).getChecksum()
+                        );
+                    }
+                    storeDirectory.copyFrom(tempRemoteSegmentDirectory, file, file, IOContext.DEFAULT);
+                    downloadedSegments.add(file);
+                    if (file.startsWith(segmentInfosSnapshotPrefix)) {
+                        assert segmentInfosSnapshotFilename == null : "There should be only one SegmentInfosSnapshot file";
+                        segmentInfosSnapshotFilename = file;
+                    }
+                } else {
+                    logger.info("file " + file + " is not able to download, adding to skipped list");
+                    skippedSegments.add(file);
+                }
+            }
+
+            if (segmentInfosSnapshotFilename != null) {
+                if (remoteSegmentStoreDirectory != null) {
+                    remoteSegmentStoreDirectory.uploadMetadata(uploadedSegments.keySet(), storeDirectory, primaryTerm, commitGeneration);
+                }
+                try (
+                    ChecksumIndexInput indexInput = new BufferedChecksumIndexInput(
+                        storeDirectory.openInput(segmentInfosSnapshotFilename, IOContext.DEFAULT)
+                    )
+                ) {
+                    SegmentInfos infosSnapshot = SegmentInfos.readCommit(store.directory(), indexInput, commitGeneration);
+                    long processedLocalCheckpoint = Long.parseLong(infosSnapshot.getUserData().get(LOCAL_CHECKPOINT_KEY));
+                    if (remoteStore != null) {
+                        store.commitSegmentInfos(infosSnapshot, processedLocalCheckpoint, processedLocalCheckpoint);
+                    }
+                    else {
+                        store.directory().sync(infosSnapshot.files(true));
+                        store.directory().syncMetaData();
```

Comment:
what is the need for doing this ? May be add reason as well as a comment.

Reply:
this is the case segrep to docrep shard case.

Reply:
yep, now, commit will be done after refresh, we are not commiting segment info file separately.